core/src/main/java/org/apache/iceberg/schema/UnionByNameVisitor.java

```diff
@@ -180,6 +180,23 @@ private void updateColumn(Types.NestedField field, Types.NestedField existingFie
     }
   }
 
+  private boolean isIgnorableTypeUpdate(Type existingType, Type newType) {
+    if (existingType.isPrimitiveType()) {
+      /* TypeUtil.isPromotionAllowed is used to check whether type promotion is allowed in the reverse order,
```

Comment:
thanks, the comment makes things much clearer. Could you please switch the comment style from /*...*/ to // as the project doesn't generally use /*...*/ for in-line comments?